pandas/core/reshape.py

```diff
@@ -202,10 +202,13 @@ def get_new_columns(self):
                 return self.removed_level
 
             lev = self.removed_level
-            vals = np.insert(lev.astype('object'), 0,
+            vals = np.insert(lev.values.astype('object'), 0,
                              _get_na_value(lev.dtype.type))
 
-            return lev._shallow_copy(vals)
+            try:
+                return lev._shallow_copy(vals.astype(lev.dtype, copy=False))
+            except ValueError:
```

Comment:
what does this fail on? why don't you always just do ``Index(vals.astype(.......), **lev._get_attributes_dict())``? e.g. pass the new values to ``Index`` and let it figure out the correct index type (with the exsting attributes). This is a very common idiom.